Spec/RestClient.swift

```diff
@@ -1698,6 +1698,85 @@ class RestClient: QuickSpec {
                 }
             }
 
+            // RSC7c
+            context("request IDs") {
+
+                it("should add 'request_id' query parameter") {
+                    let options = ARTClientOptions(key: "xxxx:xxxx")
+                    options.addRequestIds = true
+
+                    let restA = ARTRest(options: options)
+                    let mockHttpExecutor = MockHTTPExecutor()
+                    restA.internal.httpExecutor = mockHttpExecutor
+                    waitUntil(timeout: testTimeout) { done in
+                        restA.channels.get("foo").publish(nil, data: "something") { error in
+                            expect(error).to(beNil())
+                            guard let url = mockHttpExecutor.requests.first?.url else {
+                                fail("No requests found")
+                                return
+                            }
+                            expect(url.query).to(contain("request_id"))
+                            done()
+                        }
+                    }
+
+                    mockHttpExecutor.reset()
+
+                    options.addRequestIds = false
+                    let restB = ARTRest(options: options)
+                    restB.internal.httpExecutor = mockHttpExecutor
+                    waitUntil(timeout: testTimeout) { done in
+                        restB.channels.get("foo").publish(nil, data: "something") { error in
+                            expect(error).to(beNil())
+                            guard let url = mockHttpExecutor.requests.first?.url else {
```

Comment:
Given you're reliant on the implementation of the `mockHttpExecutor.reset()` call above, I would perhaps assert at this point that there is only one request.

Reply:
Yap, makes sense. Done 33b1356.